models/Channel.php

```diff
@@ -78,8 +81,226 @@ class Channel extends Model
     public $hasNew = false;
 
     /**
-     * Apply embed code to channel.
+     * Handler for the pages.menuitem.getTypeInfo event.
+     * Returns a menu item type information. The type information is returned as array
+     * with the following elements:
+     * - references - a list of the item type reference options. The options are returned in the
+     *   ["key"] => "title" format for options that don't have sub-options, and in the format
+     *   ["key"] => ["title"=>"Option title", "items"=>[...]] for options that have sub-options. Optional,
+     *   required only if the menu item type requires references.
+     * - nesting - Boolean value indicating whether the item type supports nested items. Optional,
+     *   false if omitted.
+     * - dynamicItems - Boolean value indicating whether the item type could generate new menu items.
+     *   Optional, false if omitted.
+     * - cmsPages - a list of CMS pages (objects of the Cms\Classes\Page class), if the item type requires a CMS page reference to
+     *   resolve the item URL.
+     * @param string $type Specifies the menu item type
+     * @return array Returns an array
      */
+    public static function getMenuTypeInfo($type)
+    {
+        $result = [];
+
+        if ($type == 'forum-channel') {
+            $result = [
+                'references' => self::listChildrenOptions(),
+                'nesting' => true,
+                'dynamicItems' => true
```

Comment:
@RomainMazB where possible, please ensure that the last option in an array has a trailing comma. It helps with file diffs.
```suggestion
                'dynamicItems' => true,
```